exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java

```diff
@@ -447,25 +447,41 @@ private static void writeAttributePairs(
     try {
       attributes.forEach(
```

Comment:
Can we add a comment that says that this logic depends on iteration being sorted? We could actually compare each key to the `previousKey` and confirms this assumption. If not sorted, we could log a warning, or even transform the incoming non-sorted attributes to a sorted version before continuing.

Reply:
Added a comment.  In java, is there a way to log without spamming users here?  It is easy to compare the previous key to the current and log, but that would log an enormous amount if someone used a non-sorted attribute implementation.  Given that key collisions are expected to be rare that doesn't seem ideal.  I also looked through the attribute interface, and didn't see a way to sort it unless it was the ArrayBackedAttributes implementation.